### ISSUE TITLE
Optimisation Scheduler

### ISSUE THREAD

**Issue by okpatil4u** [huggingface/candle#685]
Hello @LaurentMazare, I couldn't find any optimization scheduler in the library. Is there an alternative package available which I could readily use ? Or do I have to write it manually by accessing state of optimizer object ?

For example, I have written llama.c from scratch. But my loss increases after gradually reducing to 2 to 3 range. I think I have to introduce the scheduler. Hence the question.

**Comment by LaurentMazare** [comment]
Great that you're trying the llama.c from scratch, I've just added some hooks so that the learning rate can be set on optimizers via `.set_learning_rate(new_lr)`. For now I would suggest doing it manually on your side using these and based on the lr schedule of your liking. If you get it to work well, it would be great to look at your code and see if there are bits that we want to integrate in the optimizer modules.
Also fwiw, there is some llama.c training bits in [our examples](https://github.com/huggingface/candle/blob/main/candle-examples/examples/llama2-c/training.rs), they seem to be learning without the schedule but it's likely to improve things there too!

**Comment by okpatil4u** [comment]
Thanks Laurent, you are a saint.

I will try it out and let you know.
